torch_audiomentations/augmentations/spliceout.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from turtle import forward
+import torch
+from typing import Optional
+from torch import Tensor
+
+from ..core.transforms_interface import BaseWaveformTransform
+from ..utils.dsp import convert_decibels_to_amplitude_ratio
+from ..utils.object_dict import ObjectDict
+
+
+class SpliceOut(BaseWaveformTransform):
+
+    """
+    spliceout augmentation proposed in https://arxiv.org/pdf/2110.00046.pdf
+    silence padding is added at the end to retain the audio length.
+    """
+
+    supported_modes = {"per_batch", "per_example"}
+
+    def __init__(
+        self,
+        num_time_intervals,
+        max_width,
```

Comment:
Most transforms in audiomentations and torch-audiomentations provide sane defaults as a good starting point, and I'd love to see that here too.

Based on what I read in the paper I would suggest num_time_intervals = 8 and max_width = 400 ms as default. The problem with max_width is that it is currently defined as the number of samples, so it will change based on sample rate. Maybe it makes more sense to specify it in seconds by default? Other transforms, like `Shift`, have an extra parameter that defines the unit (samples, seconds or fraction). Maybe that's a useful pattern to adopt here as well?

https://github.com/asteroid-team/torch-audiomentations/blob/master/torch_audiomentations/augmentations/shift.py#L81